File: Source/src/todo.cpp
#include "../headers/todo.h"


/*
 * Sets the values of the todo and the status into the default string of every todo
*/
Todo::Todo(std::string name,std::string start_date,std::string end_date,std::string date_created,int id, std::string tag){
        this->editTodo(name,start_date,end_date,tag);
        this->status="Not Started";
        this->id=id;
        this->date_created=date_created;
        this->tag=tag;
}
/*
 * Edits the values of the todo
*/
void Todo::editTodo(std::string name,std::string start_date,std::string end_date, std::string tag){
		this->name=name;
		this->start_date=start_date;
		this->end_date=end_date;
        this->tag=tag;
}
/*
 * Getter for the name
*/
std::string Todo::getName(){
		return this->name;
}
/*
 * Getter for the start date
*/
std::string Todo::getStartDate(){
		return this->start_date;
}
/*
 * Getter for the end date
*/
std::string Todo::getEndDate(){
		return this->end_date;
}
/*
 * Getter for the date created
*/
std::string Todo::getDateCreated(){
		return this->date_created;
}
/*
 * Updates the status of the todo
*/
void Todo::updateStatus(std::string status){
		this->status=status;
}
/*
 * Getter for the id
*/
int Todo::getId(){
		return this->id;
}
/*
 * Getter for the status
*/
std::string Todo::getStatus(){
		return this->status;
}
/*
 * Getter for the status
*/
std::string Todo::getTag(){
        return this->tag;
}
/*
 * Creates a string with the values of the todo and returns it
*/
std::string Todo::toString(){
        std::string s;
        s=name+"  "+start_date+"  "+end_date+"  "+date_created+"  "+status+" "+tag;
        //s=std::to_string(id)+"{Delimiter}"+name+"{Delimiter}"+start_date+"{Delimiter}"+end_date+"{Delimiter}"+date_created+"{Delimiter}"+status;
        return s;
}
/*
 * Default destructor
*/
Todo::~Todo(){
		//Write entry to history
}

File: src/todo.py
# This Python file uses the following encoding: utf-8

class Todo:
    def __init__(self, name: str, start_date: str, end_date: str, date_created: str, id: int, tag: str):
        self.editTodo(name, start_date, end_date, tag)
        self.status = "Not Started"
        self.id = id
        self.date_created = date_created
        self.tag = tag
        self.date_completed=None

    def editTodo(self, name: str, start_date: str, end_date: str, tag: str):
        self.name = name
        self.start_date = start_date
        self.end_date = end_date
        self.tag = tag

    def getName(self) -> str:
        return self.name

    def getStartDate(self) -> str:
        return self.start_date

    def getEndDate(self) -> str:
        return self.end_date

    def getDateCreated(self) -> str:
        return self.date_created

    def updateStatus(self, status: str):
        self.status = status

    def getId(self) -> int:
        return self.id

    def getStatus(self) -> str:
        return self.status

    def getTag(self) -> str:
        return self.tag

    def toString(self) -> str:
        s = f"{self.name}  {self.start_date}  {self.end_date}  {self.date_created}  {self.status} {self.tag}"
        return s

    def __del__(self):
        # Write entry to history
        pass

